src/libsyntax/parse/parser.rs

```diff
@@ -5963,6 +5974,37 @@ impl<'a> Parser<'a> {
                                     maybe_append(attrs, extra_attrs));
             return Ok(Some(item));
         }
+        if self.eat_keyword(keywords::Auto) {
```

Comment:
This is not backward compatible, we should go here only if `auto` is followed by `trait`, see how `union` items are parsed for a similar example.